src/Corax/Queries/MultiTermMatch.cs

```diff
@@ -13,33 +14,47 @@ namespace Corax.Queries
     public unsafe struct MultiTermMatch<TTermProvider> : IQueryMatch
         where TTermProvider : ITermProvider
     {
-        internal long _totalResults;
-        internal long _current;
-        internal long _currentIdx;
+        private readonly bool _isBoosting;
+        private long _totalResults;
+        private long _current;
+        private long _currentIdx;
         private QueryCountConfidence _confidence;
+
+        private Bm25Relevance* _frequencyPerFieldStartPtr;
+        private int _currentFreqIdx;
+        private int _frequenciesSize;
+        private IDisposable _handler;
+        
         
         internal TTermProvider _inner;
         private TermMatch _currentTerm;        
         private readonly ByteStringContext _context;
 
-        public bool IsBoosting => false;
+        public bool IsBoosting => _isBoosting;
         public long Count => _totalResults;
         public long Current => _currentIdx <= QueryMatch.Start ? _currentIdx : _current;
 
         public QueryCountConfidence Confidence => _confidence;
 
-        public MultiTermMatch(ByteStringContext context, TTermProvider inner, long totalResults = 0, QueryCountConfidence confidence = QueryCountConfidence.Low)
+        public MultiTermMatch(in FieldMetadata field, ByteStringContext context, TTermProvider inner, long totalResults = 0, QueryCountConfidence confidence = QueryCountConfidence.Low)
         {
             _inner = inner;
             _context = context;
             _current = QueryMatch.Start;
             _currentIdx = QueryMatch.Start;
             _totalResults = totalResults;
             _confidence = confidence;
-
             var result = _inner.Next(out _currentTerm);
             if (result == false)
                 _current = QueryMatch.Invalid;
+            
+            _isBoosting = field.HasBoost;
+            if (_isBoosting)
+            {
+                _handler = _context.Allocate(64 * Unsafe.SizeOf<Bm25Relevance>(), out var bufferOutput);
+                _frequencyPerFieldStartPtr = (Bm25Relevance*)bufferOutput.Ptr;
+                _frequenciesSize = 64;
+            }
```

Comment:
Since `BM25Relevance`  holds `IDisposable` for memory I cannot save this into the buffer from context.

I got warning
`This takes the address of, gets the size of, or declares a pointer to a managed type ('Bm25Relevance')`

What is the best way to handle that? It's gonna be released at the end of a transaction but I think it's better to release it immediately when we don't need it.

Reply:
Per match